diff -urN sdcc.orig/sdas/as6808/Makefile.in sdcc/sdas/as6808/Makefile.in
--- sdcc.orig/sdas/as6808/Makefile.in	2010-03-23 21:59:47.000000000 +0100
+++ sdcc/sdas/as6808/Makefile.in	2010-06-16 16:05:12.000000000 +0200
@@ -36,7 +36,7 @@
 
 CPPFLAGS        = @CPPFLAGS@ -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
 
 OBJDIR = obj
 
diff -urN sdcc.orig/sdas/as8051/Makefile.in sdcc/sdas/as8051/Makefile.in
--- sdcc.orig/sdas/as8051/Makefile.in	2010-03-23 21:59:47.000000000 +0100
+++ sdcc/sdas/as8051/Makefile.in	2010-06-16 16:05:12.000000000 +0200
@@ -36,7 +36,7 @@
 
 CPPFLAGS        = @CPPFLAGS@ -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
 
 OBJDIR = obj
 
diff -urN sdcc.orig/sdas/asgb/Makefile.in sdcc/sdas/asgb/Makefile.in
--- sdcc.orig/sdas/asgb/Makefile.in	2010-03-23 21:59:47.000000000 +0100
+++ sdcc/sdas/asgb/Makefile.in	2010-06-16 16:05:12.000000000 +0200
@@ -36,7 +36,7 @@
 
 CPPFLAGS        = @CPPFLAGS@ -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
 
 OBJDIR = obj
 
diff -urN sdcc.orig/sdas/asz80/Makefile.in sdcc/sdas/asz80/Makefile.in
--- sdcc.orig/sdas/asz80/Makefile.in	2010-03-23 21:59:47.000000000 +0100
+++ sdcc/sdas/asz80/Makefile.in	2010-06-16 16:05:12.000000000 +0200
@@ -36,7 +36,7 @@
 
 CPPFLAGS        = @CPPFLAGS@ -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
 
 OBJDIR = obj
 
