diff -Naur sdcc.orig/Makefile.common.in sdcc.mod2/Makefile.common.in
--- sdcc.orig/Makefile.common.in	2003-08-01 10:49:45.000000000 +0200
+++ sdcc.mod2/Makefile.common.in	2004-04-19 01:18:11.159578658 +0200
@@ -30,6 +30,7 @@
 infodir         = @infodir@
 srcdir          = @srcdir@
 docdir          = @docdir@
+DESTDIR		= 
 
 include_dir_suffix = @include_dir_suffix@
 lib_dir_suffix     = @lib_dir_suffix@
diff -Naur sdcc.orig/as/Makefile sdcc.mod2/as/Makefile
--- sdcc.orig/as/Makefile	2004-01-26 09:05:15.000000000 +0100
+++ sdcc.mod2/as/Makefile	2004-04-19 01:19:52.231855267 +0200
@@ -10,16 +10,16 @@
 	$(MAKE) -C hc08 all E=$(E) BUILDDIR=$(BUILDDIR)
 
 install: all install-doc
-	$(INSTALL) $(PRJDIR)/bin/as-z80$(EXEEXT) `echo $(bindir)/as-z80$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/as-z80$(EXEEXT)|sed '$(transform)'`
-	$(INSTALL) $(PRJDIR)/bin/as-gbz80$(EXEEXT) `echo $(bindir)/as-gbz80$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/as-gbz80$(EXEEXT)|sed '$(transform)'`
-	$(INSTALL) $(PRJDIR)/bin/as-hc08$(EXEEXT) `echo $(bindir)/as-hc08$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/as-hc08$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(PRJDIR)/bin/as-z80$(EXEEXT) $(DESTDIR)`echo $(bindir)/as-z80$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/as-z80$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(PRJDIR)/bin/as-gbz80$(EXEEXT) $(DESTDIR)`echo $(bindir)/as-gbz80$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/as-gbz80$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(PRJDIR)/bin/as-hc08$(EXEEXT) $(DESTDIR)`echo $(bindir)/as-hc08$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/as-hc08$(EXEEXT)|sed '$(transform)'`
 
 install-doc:
-	$(INSTALL) -d $(docdir)/aslink
-	cp -f `find doc -maxdepth 1 -type f` $(docdir)/aslink
+	$(INSTALL) -d $(DESTDIR)$(docdir)/aslink
+	cp -f `find doc -maxdepth 1 -type f` $(DESTDIR)$(docdir)/aslink
 
 uninstall:
 	cd $(bindir); rm -f as-z80$(EXEEXT) as-gbz80$(EXEEXT) as-hc08$(EXEEXT)
diff -Naur sdcc.orig/as/mcs51/Makefile.in sdcc.mod2/as/mcs51/Makefile.in
--- sdcc.orig/as/mcs51/Makefile.in	2003-08-22 22:17:12.000000000 +0200
+++ sdcc.mod2/as/mcs51/Makefile.in	2004-04-19 01:20:56.552031054 +0200
@@ -63,10 +63,10 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	$(INSTALL) $(ASX8051) `echo $(bindir)/asx8051$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/asx8051$(EXEEXT)|sed '$(transform)'`
-	$(INSTALL) $(ASLINK) `echo $(bindir)/aslink$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/aslink$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(ASX8051) $(DESTDIR)`echo $(bindir)/asx8051$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/asx8051$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(ASLINK) $(DESTDIR)`echo $(bindir)/aslink$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/aslink$(EXEEXT)|sed '$(transform)'`
 
 # Deleting all the installed files
 # --------------------------------
@@ -87,7 +87,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	$(INSTALL) -d $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/as/z80/Makefile.in sdcc.mod2/as/z80/Makefile.in
--- sdcc.orig/as/z80/Makefile.in	2001-08-16 22:04:20.000000000 +0200
+++ sdcc.mod2/as/z80/Makefile.in	2004-04-19 01:21:47.576617426 +0200
@@ -47,7 +47,7 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	$(INSTALL) $(ASXZ80) $(bindir)/as-z80
+	$(INSTALL) $(ASXZ80) $(DESTDIR)$(bindir)/as-z80
 
 # Deleting all the installed files
 # --------------------------------
@@ -68,7 +68,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	$(INSTALL) -d $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/configure.in sdcc.mod2/configure.in
--- sdcc.orig/configure.in	2004-01-14 00:18:32.000000000 +0100
+++ sdcc.mod2/configure.in	2004-04-19 12:12:11.628229523 +0200
@@ -40,6 +40,7 @@
 AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_CPP
+AC_PROG_CXXCPP
 AC_PROG_INSTALL
 AC_PROG_RANLIB
 AC_PROG_LEX
@@ -649,7 +650,7 @@
 # ===========================================================================
 AC_CONFIG_SUBDIRS(support/cpp2)
 AC_CONFIG_SUBDIRS(packihx)
-AC_CONFIG_SUBDIRS(sim/ucsim)
+#AC_CONFIG_SUBDIRS(sim/ucsim)
 #AC_CONFIG_SUBDIRS(device/lib/pic16)
 
 AC_CONFIG_FILES([main.mk:main_in.mk
diff -Naur sdcc.orig/debugger/mcs51/Makefile.in sdcc.mod2/debugger/mcs51/Makefile.in
--- sdcc.orig/debugger/mcs51/Makefile.in	2003-04-24 23:06:27.000000000 +0200
+++ sdcc.mod2/debugger/mcs51/Makefile.in	2004-04-19 01:23:07.602126768 +0200
@@ -56,10 +56,10 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	$(INSTALL) $(TARGET) $(bindir)/sdcdb$(EXEEXT)
-	$(STRIP) $(bindir)/sdcdb$(EXEEXT)
-	cp $(PRJDIR)/debugger/mcs51/sdcdb.el $(bindir)/sdcdb.el
-	cp $(PRJDIR)/debugger/mcs51/sdcdbsrc.el $(bindir)/sdcdbsrc.el
+	$(INSTALL) $(TARGET) $(DESTDIR)$(bindir)/sdcdb$(EXEEXT)
+	$(STRIP) $(DESTDIR)$(bindir)/sdcdb$(EXEEXT)
+	cp $(PRJDIR)/debugger/mcs51/sdcdb.el $(DESTDIR)$(bindir)/sdcdb.el
+	cp $(PRJDIR)/debugger/mcs51/sdcdbsrc.el $(DESTDIR)$(bindir)/sdcdbsrc.el
 
 
 # Deleting all the installed files
@@ -82,7 +82,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	$(INSTALL) -d $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/device/include/Makefile.in sdcc.mod2/device/include/Makefile.in
--- sdcc.orig/device/include/Makefile.in	2004-02-21 20:48:05.000000000 +0100
+++ sdcc.mod2/device/include/Makefile.in	2004-04-19 01:24:59.110296904 +0200
@@ -40,9 +40,9 @@
 # Compiling and installing everything and running test
 # ----------------------------------------------------
 install: all installdirs
-	$(CP) -r asm *.h $(sdcc_includedir)
+	$(CP) -r asm *.h $(DESTDIR)$(sdcc_includedir)
 	if [ "`grep pic16 ../../ports.build`" = pic16 ]; then \
-		$(CP) pic16/*.h $(sdcc_includedir)/pic16 ; \
+		$(CP) pic16/*.h $(DESTDIR)$(sdcc_includedir)/pic16 ; \
 	fi
 	rm -rf `find $(sdcc_includedir) -type d -name 'CVS'`
 
@@ -66,8 +66,8 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	[ -d $(sdcc_includedir) ] || mkdir -p $(sdcc_includedir)
-	[ -d $(sdcc_includedir)/pic16 ] || mkdir -p $(sdcc_includedir)/pic16
+	[ -d $(DESTDIR)$(sdcc_includedir) ] || mkdir -p $(DESTDIR)$(sdcc_includedir)
+	[ -d $(DESTDIR)$(sdcc_includedir)/pic16 ] || mkdir -p $(DESTDIR)$(sdcc_includedir)/pic16
 
 
 # Creating dependencies
diff -Naur sdcc.orig/device/lib/Makefile.in sdcc.mod2/device/lib/Makefile.in
--- sdcc.orig/device/lib/Makefile.in	2004-01-13 16:00:08.000000000 +0100
+++ sdcc.mod2/device/lib/Makefile.in	2004-04-19 01:27:36.596589719 +0200
@@ -240,9 +240,9 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	cp -r $(BUILDDIR)/* $(sdcc_libdir)
-	cp -r ds390 ds400 gbz80 z80 hc08 *.c $(sdcc_libdir)/src
-	rm -r `find $(sdcc_libdir)/src -name '*.rel' -or -name '*.dump*' -or -name '*.sym' -or -name '*.o' -or -name '*.lst' -or -name '*.asm' -or -name 'CVS'`
+	cp -r $(BUILDDIR)/* $(DESTDIR)$(sdcc_libdir)
+	cp -r ds390 ds400 gbz80 z80 hc08 *.c $(DESTDIR)$(sdcc_libdir)/src
+	rm -r `find $(DESTDIR)$(sdcc_libdir)/src -name '*.rel' -or -name '*.dump*' -or -name '*.sym' -or -name '*.o' -or -name '*.lst' -or -name '*.asm' -or -name 'CVS'`
 
 # Deleting all the installed files
 # --------------------------------
@@ -262,16 +262,16 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	[ -d $(sdcc_libdir) ] || mkdir -p $(sdcc_libdir)
+	[ -d $(DESTDIR)$(sdcc_libdir) ] || mkdir -p $(DESTDIR)$(sdcc_libdir)
 	for model in $(MODELS); do \
-	 [ -d $(sdcc_libdir)/$$model ] || \
-	 mkdir -p $(sdcc_libdir)/$$model; \
+	 [ -d $(DESTDIR)$(sdcc_libdir)/$$model ] || \
+	 mkdir -p $(DESTDIR)$(sdcc_libdir)/$$model; \
 	done
-	[ -d $(sdcc_libdir)/ds390 ] || mkdir -p $(sdcc_libdir)/ds390
-	[ -d $(sdcc_libdir)/ds400 ] || mkdir -p $(sdcc_libdir)/ds400	
-	[ -d $(sdcc_libdir)/hc08 ] || mkdir -p $(sdcc_libdir)/hc08	
-	[ -f $(sdcc_libdir)/pic16 ] || mkdir -p $(sdcc_libdir)/pic16
-	mkdir -p $(sdcc_libdir)/src
+	[ -d $(DESTDIR)$(sdcc_libdir)/ds390 ] || mkdir -p $(DESTDIR)$(sdcc_libdir)/ds390
+	[ -d $(DESTDIR)$(sdcc_libdir)/ds400 ] || mkdir -p $(DESTDIR)$(sdcc_libdir)/ds400	
+	[ -d $(DESTDIR)$(sdcc_libdir)/hc08 ] || mkdir -p $(DESTDIR)$(sdcc_libdir)/hc08	
+	[ -f $(DESTDIR)$(sdcc_libdir)/pic16 ] || mkdir -p $(DESTDIR)$(sdcc_libdir)/pic16
+	mkdir -p $(DESTDIR)$(sdcc_libdir)/src
 
 # Creating dependencies
 # ---------------------
diff -Naur sdcc.orig/doc/Makefile sdcc.mod2/doc/Makefile
--- sdcc.orig/doc/Makefile	2004-01-31 23:35:46.000000000 +0100
+++ sdcc.mod2/doc/Makefile	2004-04-20 05:21:21.886276246 +0200
@@ -9,16 +9,16 @@
 	$(CDB).pdf $(CDB).html/index.html $(CDB).txt
 
 install:
-	$(INSTALL) -d $(docdir)
-	cp -rf *.txt z80 avr $(docdir)
-	if [ -f $(MANUAL).html/index.html ]; then cp -rf $(MANUAL).html $(docdir); fi
-	if [ -f $(TSS).html/index.html    ]; then cp -rf $(TSS).html    $(docdir); fi
-	if [ -f $(CDB).html/index.html    ]; then cp -rf $(CDB).html    $(docdir); fi
-	if [ -f $(MANUAL).pdf             ]; then cp -rf $(MANUAL).pdf  $(docdir); fi
-	if [ -f $(TSS).pdf                ]; then cp -rf $(TSS).pdf     $(docdir); fi
-	if [ -f $(CDB).pdf                ]; then cp -rf $(CDB).pdf     $(docdir); fi
-	rm -rf `find $(docdir) -name CVS -type d`
-	rm -f $(docdir)/*/WARNINGS $(docdir)/*/*.pl $(docdir)/*/images.*
+	$(INSTALL) -d $(DESTDIR)$(docdir)
+	cp -rf *.txt z80 avr $(DESTDIR)$(docdir)
+	if [ -f $(MANUAL).html/index.html ]; then cp -rf $(MANUAL).html $(DESTDIR)$(docdir); fi
+	if [ -f $(TSS).html/index.html    ]; then cp -rf $(TSS).html    $(DESTDIR)$(docdir); fi
+	if [ -f $(CDB).html/index.html    ]; then cp -rf $(CDB).html    $(DESTDIR)$(docdir); fi
+	if [ -f $(MANUAL).pdf             ]; then cp -rf $(MANUAL).pdf  $(DESTDIR)$(docdir); fi
+	if [ -f $(TSS).pdf                ]; then cp -rf $(TSS).pdf     $(DESTDIR)$(docdir); fi
+	if [ -f $(CDB).pdf                ]; then cp -rf $(CDB).pdf     $(DESTDIR)$(docdir); fi
+	rm -rf `find $(DESTDIR)$(docdir) -name CVS -type d`
+	rm -f $(DESTDIR)$(docdir)/*/WARNINGS $(DESTDIR)$(docdir)/*/*.pl $(DESTDIR)$(docdir)/*/images.*
 
 uninstall:
 	rm -rf $(docdir)
diff -Naur sdcc.orig/link/Makefile sdcc.mod2/link/Makefile
--- sdcc.orig/link/Makefile	2003-04-24 23:06:31.000000000 +0200
+++ sdcc.mod2/link/Makefile	2004-04-19 01:30:13.977893244 +0200
@@ -8,10 +8,10 @@
 	$(MAKE) -C z80 _link-z80 _link-gbz80 E=$(E) BUILDDIR=../../bin/
 
 install: all
-	$(INSTALL) $(PRJDIR)/bin/link-z80$(EXEEXT) `echo $(bindir)/link-z80$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/link-z80$(EXEEXT)|sed '$(transform)'`
-	$(INSTALL) $(PRJDIR)/bin/link-gbz80$(EXEEXT) `echo $(bindir)/link-gbz80$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/link-gbz80$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(PRJDIR)/bin/link-z80$(EXEEXT) $(DESTDIR)`echo $(bindir)/link-z80$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/link-z80$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(PRJDIR)/bin/link-gbz80$(EXEEXT) $(DESTDIR)`echo $(bindir)/link-gbz80$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/link-gbz80$(EXEEXT)|sed '$(transform)'`
 
 uninstall:
 	cd $(bindir); rm -f link-z80$(EXEEXT) link-gbz80$(EXEEXT)
diff -Naur sdcc.orig/link/z80/Makefile.in sdcc.mod2/link/z80/Makefile.in
--- sdcc.orig/link/z80/Makefile.in	2003-04-24 23:06:33.000000000 +0200
+++ sdcc.mod2/link/z80/Makefile.in	2004-04-19 01:30:49.339141737 +0200
@@ -54,7 +54,7 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	$(INSTALL) $(LKZ80) $(bindir)/link-z80$(EXEEXT)
+	$(INSTALL) $(LKZ80) $(DESTDIR)$(bindir)/link-z80$(EXEEXT)
 
 # Deleting all the installed files
 # --------------------------------
@@ -75,7 +75,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	$(INSTALL) -d $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/packihx/Makefile.in sdcc.mod2/packihx/Makefile.in
--- sdcc.orig/packihx/Makefile.in	2003-04-24 23:06:34.000000000 +0200
+++ sdcc.mod2/packihx/Makefile.in	2004-04-19 01:31:24.773382464 +0200
@@ -13,8 +13,8 @@
 all: $(TARGET)
 
 install: all installdirs
-	$(INSTALL) $(TARGET) `echo $(bindir)/packihx$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/packihx$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(TARGET) $(DESTDIR)`echo $(bindir)/packihx$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/packihx$(EXEEXT)|sed '$(transform)'`
 
 # Deleting all the installed files
 # --------------------------------
@@ -34,7 +34,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	$(INSTALL) -d $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/sim/ucsim/avr.src/Makefile.in sdcc.mod2/sim/ucsim/avr.src/Makefile.in
--- sdcc.orig/sim/ucsim/avr.src/Makefile.in	2003-04-24 23:06:42.000000000 +0200
+++ sdcc.mod2/sim/ucsim/avr.src/Makefile.in	2004-04-19 01:32:12.700297792 +0200
@@ -65,7 +65,7 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	$(INSTALL) -s savr$(EXEEXT) $(bindir)
+	$(INSTALL) -s savr$(EXEEXT) $(DESTDIR)$(bindir)
 
 
 # Deleting all the installed files
@@ -89,7 +89,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	test -d $(bindir) || $(INSTALL) -d $(bindir)
+	test -d $(DESTDIR)$(bindir) || $(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/sim/ucsim/doc/Makefile.in sdcc.mod2/sim/ucsim/doc/Makefile.in
--- sdcc.orig/sim/ucsim/doc/Makefile.in	2003-02-14 17:27:51.000000000 +0100
+++ sdcc.mod2/sim/ucsim/doc/Makefile.in	2004-04-19 01:33:29.009201936 +0200
@@ -34,10 +34,10 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	cp *.html $(docdir)
-	cp *.gif $(docdir)
-	cp *.jpg $(docdir)
-	cp *.fig $(docdir)
+	cp *.html $(DESTDIR)$(docdir)
+	cp *.gif $(DESTDIR)$(docdir)
+	cp *.jpg $(DESTDIR)$(docdir)
+	cp *.fig $(DESTDIR)$(docdir)
 
 
 # Deleting all the installed files
@@ -61,7 +61,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	test -d $(docdir) || $(INSTALL) -d $(docdir)
+	test -d $(DESTDIR)$(docdir) || $(INSTALL) -d $(DESTDIR)$(docdir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/sim/ucsim/s51.src/Makefile.in sdcc.mod2/sim/ucsim/s51.src/Makefile.in
--- sdcc.orig/sim/ucsim/s51.src/Makefile.in	2003-04-24 23:06:47.000000000 +0200
+++ sdcc.mod2/sim/ucsim/s51.src/Makefile.in	2004-04-19 01:34:47.342891169 +0200
@@ -68,7 +68,7 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	$(INSTALL) -s s51$(EXEEXT) $(bindir)
+	$(INSTALL) -s s51$(EXEEXT) $(DESTDIR)$(bindir)
 
 
 # Deleting all the installed files
@@ -94,7 +94,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	test -d $(bindir) || $(INSTALL) -d $(bindir)
+	test -d $(DESTDIR)$(bindir) || $(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/sim/ucsim/z80.src/Makefile.in sdcc.mod2/sim/ucsim/z80.src/Makefile.in
--- sdcc.orig/sim/ucsim/z80.src/Makefile.in	2003-04-24 23:06:52.000000000 +0200
+++ sdcc.mod2/sim/ucsim/z80.src/Makefile.in	2004-04-19 01:36:58.979925000 +0200
@@ -72,7 +72,7 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	$(INSTALL) -s sz80$(EXEEXT) $(bindir)
+	$(INSTALL) -s sz80$(EXEEXT) $(DESTDIR)$(bindir)
 
 
 # Deleting all the installed files
@@ -96,7 +96,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	test -d $(bindir) || $(INSTALL) -d $(bindir)
+	test -d $(DESTDIR)$(bindir) || $(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/src/Makefile.in sdcc.mod2/src/Makefile.in
--- sdcc.orig/src/Makefile.in	2003-04-24 23:06:53.000000000 +0200
+++ sdcc.mod2/src/Makefile.in	2004-04-19 01:37:48.405681054 +0200
@@ -56,8 +56,8 @@
 # Compiling and installing everything and runing test
 # ---------------------------------------------------
 install: all installdirs
-	$(INSTALL) $(TARGET) `echo $(bindir)/sdcc$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/sdcc$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(TARGET) $(DESTDIR)`echo $(bindir)/sdcc$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/sdcc$(EXEEXT)|sed '$(transform)'`
 
 
 # Deleting all the installed files
@@ -78,7 +78,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	$(INSTALL) -d $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
diff -Naur sdcc.orig/support/cpp2/Makefile.in sdcc.mod2/support/cpp2/Makefile.in
--- sdcc.orig/support/cpp2/Makefile.in	2003-04-26 22:58:14.000000000 +0200
+++ sdcc.mod2/support/cpp2/Makefile.in	2004-04-19 01:38:29.545316217 +0200
@@ -108,9 +108,9 @@
 # End of variables for you to override.
 
 install: all
-	mkdir -p $(bindir)
-	$(INSTALL) $(TARGET) `echo $(bindir)/sdcpp$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/sdcpp$(EXEEXT)|sed '$(transform)'`
+	mkdir -p $(DESTDIR)$(bindir)
+	$(INSTALL) $(TARGET) $(DESTDIR)`echo $(bindir)/sdcpp$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/sdcpp$(EXEEXT)|sed '$(transform)'`
 
 uninstall:
 	rm -f $(bindir)/sdcpp$(EXEEXT)
diff -Naur sdcc.orig/support/librarian/Makefile sdcc.mod2/support/librarian/Makefile
--- sdcc.orig/support/librarian/Makefile	2003-08-01 10:09:56.000000000 +0200
+++ sdcc.mod2/support/librarian/Makefile	2004-04-20 05:07:21.501505881 +0200
@@ -13,8 +13,8 @@
 all: $(TARGET)
 
 install: all installdirs
-	$(INSTALL) $(TARGET) `echo $(bindir)/sdcclib$(EXEEXT)|sed '$(transform)'`
-	$(STRIP) `echo $(bindir)/sdcclib$(EXEEXT)|sed '$(transform)'`
+	$(INSTALL) $(TARGET) $(DESTDIR)`echo $(bindir)/sdcclib$(EXEEXT)|sed '$(transform)'`
+	$(STRIP) $(DESTDIR)`echo $(bindir)/sdcclib$(EXEEXT)|sed '$(transform)'`
 
 # Deleting all the installed files
 # --------------------------------
@@ -34,7 +34,7 @@
 # Creating installation directories
 # ---------------------------------
 installdirs:
-	$(INSTALL) -d $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
 
 
 # Creating dependencies
