diff -urN sdcc.orig/src/mcs51/gen.c sdcc/src/mcs51/gen.c
--- sdcc.orig/src/mcs51/gen.c	2007-07-18 17:17:51.000000000 +0200
+++ sdcc/src/mcs51/gen.c	2007-08-16 08:25:04.000000000 +0200
@@ -3336,6 +3336,9 @@
       emitcode(";", "naked function: no prologue.");
       return;
   }
+  
+  if (options.stack_probe)
+      emitcode ("lcall","__stack_probe");
 
   /* here we need to generate the equates for the
      register bank if required */
