diff -urN sdcc.orig/sim/ucsim/configure.in sdcc/sim/ucsim/configure.in
--- sdcc.orig/sim/ucsim/configure.in	2010-04-14 14:10:25.000000000 +0200
+++ sdcc/sim/ucsim/configure.in	2010-06-16 15:56:39.000000000 +0200
@@ -510,15 +510,7 @@
 
   # This is the first time when CFLAGS are set/modified!!
 
-  bor_IS_VALID_OPTION(ggdb)
-  bor_IS_VALID_OPTION(gstabs)
-  if test "$ucsim_cv_cxx_gstabs" = "yes"; then
-    gopt="-gstabs+ -g3"
-  elif test "$ucsim_cv_cxx_ggdb" = "yes"; then
-    gopt="-ggdb -g3"
-  else
-    gopt="-g"
-  fi
+  gopt="-g"
   if test "$CXXFLAGS"x = x ;then
     CXXFLAGS="$gopt"
   else
